sycl/doc/extensions/SubGroupBitmaskGroupFunctions/sub_group_bitmask_group_algorithms.asciidoc

```diff
@@ -0,0 +1,253 @@
+like `group_barrier`.
+
+`async_group_copy` methods would be valid for groups `group and `sub_group`.
+For subgroup, a new bitmasked implementation will also be required.
+
+```c++
+namespace sycl::ext::oneapi {
+template <typename Group, typename dataT>
+device_event async_group_copy(Group group, sycl::decorated_local_ptr<dataT> dest, sycl::decorated_global_ptr<dataT> src, size_t numElements);
+
+template <typename Group, typename dataT>
+device_event async_group_copy(Group group, sycl::decorated_global_ptr<dataT> dest, sycl::decorated_local_ptr<dataT> src, size_t numElements);
+
+template <typename Group, typename dataT>
+device_event async_group_copy(Group group, sycl::decorated_local_ptr<dataT> dest, sycl::decorated_global_ptr<dataT> src, size_t numElements, size_t srcStride);
+
+template <typename Group, typename dataT>
+device_event async_group_copy(Group group, sycl::decorated_global_ptr<dataT> dest, sycl::decorated_local_ptr<dataT> src, size_t numElements, size_t destStride);
+
+template <typename dataT>
+device_event async_group_copy(sycl::sub_group group, sycl::decorated_local_ptr<dataT> dest, sycl::decorated_global_ptr<dataT> src, size_t numElements, sycl::sub_group_mask mask);
+
+template <typename dataT>
+device_event async_group_copy(sycl::sub_group group, sycl::decorated_global_ptr<dataT> dest, sycl::decorated_local_ptr<dataT> src, size_t numElements, sycl::sub_group_mask mask);
+
+template <typename dataT>
+device_event async_group_copy(sycl::sub_group group, sycl::decorated_local_ptr<dataT> dest, sycl::decorated_global_ptr<dataT> src, size_t numElements, size_t srcStride, sycl::sub_group_mask mask);
+
+template <typename dataT>
+device_event async_group_copy(sycl::sub_group group, sycl::decorated_global_ptr<dataT> dest, sycl::decorated_local_ptr<dataT> src, size_t numElements, size_t destStride, sycl::sub_group_mask mask);
+} // namespace sycl::ext::oneapi
+```
+
+## Change to `device_event::wait` function
+`device_event::wait` will be modified to take the group that will be synchonised in the wait.
+This will be valid for the same groups as `async_group_copy` i.e. `group` or `sub_group`.
+An overload with a `sub_group_mask` parameter will be required for `sub_group`.
+
+```c++
+namespace sycl {
+  class device_event {
+  ...
+  public:
+    template <typename Group>
+    void ext_oneapi_wait(Group group) noexcept;
+    void ext_oneapi_wait(sycl::sub_group group, sycl::sub_group_mask mask) noexcept;
+  ...
+  };
+}  // namespace sycl
+```
+
+## Open Questions
+
+* Do the `sub_group_mask` overloads of functions also need the `sub_group` class as an argument, since this is implicit.
+* How to handle the bitmask partition not dividing the sub_group size evenly?
+* What to do if the bitmask sub_group partition doesn't divide the size of the nd_range evenly? Potentially UB.
+* Should async_group_copy destination and source be flipped to match sycl::handler::copy and sycl::handler::memcopy.
```

Comment:
I think so.

https://github.com/KhronosGroup/SYCL-Docs/pull/168 changed the SYCL 2020 specification so that if a function is called `copy` and copies type `T` it is `copy(T* src, T* dest)`, whereas if a function is called `memcpy` and copies bytes, it is `memcpy(void* dst, void* src)`.  So this should be `async_group_copy(T* src, T* dest)`.